Change the default color scheme (Closes #8)

kimikage opened the pull request:
This adds the support for hilighting with multiple colors for the runtime dispatch or gc events (i.e. `colorrt` or `colorgc`).
This also changes the default `n` in `FlameColors()` from 5 to 2 to make highlight colors more distinguishable.

Although the docs should be updated with new default colors, this PR only update the docstring of `FlameColors()`.

kimikage commented:
I really want to generate the figures when building the documentation, but one of problems is that profiling data is not reproducible. One of the solutions is to add raw graph data to the assets instead of PNG images, and visualize the graph dynamically. However, https://github.com/timholy/ProfileView.jl/issues/73 should be fixed first. I don't know the detail of the issue.

kimikage commented on src/render.jl:
```diff
@@ -47,12 +67,22 @@ function (colors::FlameColors)(s::Symbol)
+    else
+        isa(colorvec, Colorant) && return colorvec
+        n = (length(colorvec) + 1)÷2
+        idx = mod1(idx, n) + iseven(j) * n
```
> If we're going to do this, should we just get rid of the `colorsodd` and `colorseven` fields and just have `colors`?

If we want the consistency or simple implementation, we should. However, I did this for compatibility. This commit breaks the definition of `FlameColors`, but we can use its default constructor without modifications. Of course, this "redundant" code can use all of the generated vectors.

timholy replied:
FlameGraphs is quite new. Now is definitely the time to make breaking changes, and field names are generally "internal" anyway. I'd say let's make that change, if you're up for doing it. (Or I can, but it will be a few days due to other commitments.)

kimikage replied:
That makes sense. I'll change the `FlameColors`.

kimikage commented:
Just for the automatic figure generation, a custom serializer with limited functionality would be sufficient, because there is no need to work with the "real" `MethodInstance`s.

On the other hand, considering https://github.com/timholy/FlameGraphs.jl/issues/7, more drastic measures are needed, too. I also think the measures should be in an even lower level module, considering  https://github.com/davidanthoff/ProfileVega.jl/issues/5. In the long term, it is hoped that the `Profile` module will be improved but it is not so realistic given compatibility.

kimikage commented:
I changed the fields of the `FlameColors` structure.
(I made a mistake in git operation. So, when you find strange logs, please ignore them.:sweat_smile:)

Since I modified the tests  only related to the source code changes, it might be better to add more tests (e.g. passing `nothing` for `colorsrt`).

timholy commented:
Thanks! I'll tackle the rest from here later this week.
